lesson16.md

```diff
@@ -0,0 +1,114 @@
+<!-- .slide: id="lesson16" -->
+
+# Basic Frontend - Fall 2023
+
+Lesson 16, Tuesday, 2023-11-21
+
+---
+
+Today's lesson will be the last lesson before starting with the project, and we will be doing a small project to get you
+up and running!
+
+The structure is very loose, it is up to you to explore and find what you enjoy!
+
+---
+
+## Dogs
+
+https://dog.ceo/dog-api/documentation/random
+> Display a random dog image
+
+https://kinduff.github.io/dog-api/
+> Fetch dog facts
+
+---
+
+### Task
+
+Create a simple webpage, that fetches dog images from the API. Pair it with the second API and show some dog facts
+together with images.
+
+---
+
+## Holidays
+
+https://date.nager.at/Api
+> Query the holidays of over 100 countries
+
+---
+
+### Task
+
+Show the holidays in Germany for 2024.
+This differs from state to state. Add a dropdown to select a specific state?
+
+---
+
+## Maps
+
+https://leafletjs.com/examples/quick-start/
```

Comment:
Doesn't this require an API key?

Reply:
Maybe it did in the past, but not anymore.
here a small Proof of Concept, basically copying and pasting the code from the quick start tutorial:

```html
<html lang="en">
<head>
  <title>Leaflet Demo</title>
  <link rel="stylesheet" href="https://unpkg.com/leaflet@1.9.4/dist/leaflet.css"
        integrity="sha256-p4NxAoJBhIIN+hmNHrzRCf9tD/miZyoHS5obTRR9BMY="
        crossorigin="" />
  <!-- Make sure you put this AFTER Leaflet's CSS -->
  <script src="https://unpkg.com/leaflet@1.9.4/dist/leaflet.js"
          integrity="sha256-20nQCchB9co0qIjJZRGuk2/Z9VM+kNiyxNV1lvTlZBo="
          crossorigin=""></script>
  <style>
      #map {
          height: 800px;
      }
  </style>
</head>
<body>
<div id="map"></div>
<script>
  // https://leafletjs.com/examples/quick-start/
  const map = L.map("map").setView([51.505, -0.09], 13);

  L.tileLayer("https://tile.openstreetmap.org/{z}/{x}/{y}.png", {
    maxZoom: 19,
    attribution: "&copy; <a href=\"http://www.openstreetmap.org/copyright\">OpenStreetMap</a>"
  }).addTo(map);
</script>
</body>
</html>
```